backend/finance/models.py

```diff
@@ -394,3 +400,161 @@ class ReceiptDownloadToken(models.Model):
+            invoice_instance.self_county = self_details.get("self_county", invoice_instance.self_county)
+            invoice_instance.self_country = self_details.get("self_country", invoice_instance.self_country)
+
+            # Restore financial_details
+            financial_details = changes.get("financial_details", {})
+            invoice_instance.sort_code = financial_details.get("sort_code", invoice_instance.sort_code)
+            invoice_instance.account_holder_name = financial_details.get("account_holder_name", invoice_instance.account_holder_name)
+            invoice_instance.account_number = financial_details.get("account_number", invoice_instance.account_number)
+            invoice_instance.reference = financial_details.get("reference", invoice_instance.reference)
+            invoice_instance.invoice_number = financial_details.get("invoice_number", invoice_instance.invoice_number)
+            invoice_instance.vat_number = financial_details.get("vat_number", invoice_instance.vat_number)
+            invoice_instance.currency = financial_details.get("currency", invoice_instance.currency)
+            invoice_instance.discount_amount = Decimal(financial_details.get("discount_amount", invoice_instance.discount_amount))
+            invoice_instance.discount_percentage = Decimal(
+                financial_details.get("discount_percentage", invoice_instance.discount_percentage)
+            )
+
+            # Restore dates
+            dates = changes.get("dates", {})
+            invoice_instance.date_created = dates.get("date_created", invoice_instance.date_created)
+            invoice_instance.date_issued = dates.get("date_issued", invoice_instance.date_issued)
+            invoice_instance.date_due = dates.get("date_due", invoice_instance.date_due)
+
+            # Restore status
+            invoice_instance.status = changes.get("status", invoice_instance.status)
+
+            # Restore items
+            invoice_instance.items.clear()
+            items = changes.get("items", [])
+            for item_data in items:
+                item = InvoiceItem(
+                    name=item_data.get("name"),
+                    description=item_data.get("description"),
+                    is_service=item_data.get("is_service"),
+                    hours=Decimal(item_data["hours"]) if item_data.get("is_service") and item_data.get("hours") else None,
+                    price_per_hour=(
+                        Decimal(item_data["price_per_hour"]) if item_data.get("is_service") and item_data.get("price_per_hour") else None
+                    ),
+                    price=Decimal(item_data["price"]) if item_data.get("price") else None,
+                )
+                item.save()
+                invoice_instance.items.add(item)
+
+            # Restore logo a notes
+            invoice_instance.logo = changes.get("logo", invoice_instance.logo)
```

Comment:
Does this work with the logo? I can double check next time im on if you'd like

Reply:
Uploading the logo does not work, but it is a preparation for the future